### PR TITLE
External v2

### PR DESCRIPTION
**Pull request by craggy2593**
@behaghel: Updated version of the internal / external functionality (the JavaScript fix that we demonstrated). Also includes all existing changes in the previous pull request that were already reviewed. 

**Changes include**
- Internal and External functionality controlled by JavaScript. NoScript functionality included to redirect a user from the website if they have JavaScript disabled
- IP Plugin has been removed from alternative solution after it was found that plugins do not work on GitHub pages
- Open Source has been hidden from site navigation after it was decided further work was needed on it first
- Created a simple work for us page (needs more information, but is a skeleton for information to come)
- Set the blog page as the new homepage after it was suggested to us

### CODE REVIEW

**Comment by behaghel** [comment]
@craggy2593: link to the approval dashboard please?

File: _includes/scripts.html [hunk 1]
```diff
@@ -3,10 +3,19 @@
 <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.4/js/bootstrap.min.js"></script>
 <script src="{{"/scripts/main.js" | prepend: site.baseurl}}"></script>
 <script>
+  // (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
```
Comment (behaghel) [inline _includes/scripts.html]: please delete this comment

File: _layouts/noscriptdefault.html [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<!DOCTYPE html>
+<html>
+  {% include noscripthead.html %}
+    <body class="grid-1140">
+        <header class="skycom-container">
+            <div class="alpha skycom-12">
+                {% include noscriptheader.html %}
```
Comment (behaghel) [inline _layouts/noscriptdefault.html]: why including it twice?

Comment (behaghel) [inline _layouts/noscriptdefault.html]: oh noscripthead and noscriptheadER… Can we make it less confusing please? :) It's minor hence optional but welcome.

File: index.html [hunk 3]
```diff
@@ -1,45 +1,32 @@
...
-        <h2>Sky developers are doing amazing things.
-          <span class="text-muted">See for yourself.</span>
-        </h2>
-        <p class="lead text--instructions">Take a look at <a href="/blog">what's going on</a> for Developers across Sky.</p>
-      </div>
-      <div class="skycom-5">
-        <img class="img-responsive" src="http://www.sky.com/tv/static/C3DqxkQmSv8YIMTOoxx9VzIHPFVI1fZr1I2rKA2RWOx.jpg" alt="Generic placeholder image">
-      </div>
+<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.3/jquery.min.js"></script>
+<script>
```
Comment (behaghel) [inline index.html]: can we avoid the copy-paste with blog.html here?

File: internalHtml.html [hunk 4]
```diff
@@ -0,0 +1,30 @@
+---
+layout: default
+title: Blog
+---
+<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.3/jquery.min.js"></script>
+<script>
+$.ajax({
```
Comment (behaghel) [inline internalHtml.html]: copy-paste :fearful:

File: noscript.html [hunk 5]
```diff
@@ -0,0 +1,11 @@
+---
+layout: noscriptdefault
+title: Index
+---
+<script>
+  window.location = "/";
+</script>
+
+<h3>JavaScript Disabled</h3>
```
Comment (behaghel) [inline noscript.html]: have we thought about the SEO friendliness of this?

**Comment by behaghel** [comment]
reviewed all the posts in this PR and happy for them to go live.

**Comment by craggy2593** [comment]
@behaghel: Addressed your suggested changes:
- Removed commented script as now replaced with the newer Google Analytics script we are using
- Renamed `noscriptheader` to `noscriptnavigation` (also renamed `header` to `navigation` for consistency)
- Removed `blog.html` as no longer in use, and moved duplicated code to own file, and included where required (did not move the script, as they are not duplicated, there is a minor change in what they do)
- Addressed the SEO question. Found [this](https://northcutt.com/how-the-noscript-tag-impacts-seo-hint-be-very-careful/) which stated that including it in an image is a better option. This seemed the easiest option from his suggestions, as we don't require custom messages.

**Comment by behaghel** [comment]
good job. The only concern left is the copy paste which handles internal/external access. It seems that each top-level page needs that so maybe we could create a template for this kind of page that has this logic and then each specific page would inject its own bits. That way if we create a new top-level page, we just reuse this template and all will be handled consistently?

I must admit this suggestion maybe rubbish as I haven't fully analysed the current state...

**Comment by craggy2593** [comment]
@behaghel: Moved it now to it's own script file now. It's not perfect, but the original way of doing it had to be adjusted as the AJAX call would come back after the page was loaded, so it wouldn't redirect. Therefore, the option has been to send in the required parameter in to a function.

**Comment by behaghel** [comment]
I am cool with the code (even though now I think the idea of a shared template wasn't pointless).

Now when I test it locally with `jekyll serve` I see a flickering on the Home (or Blog) section: the title _Latest Posts_ first appears then disappears. Can we remove that completely?

**Comment by craggy2593** [comment]
Thanks.

Apologies, forgot to remove the 'Latest Posts' text from the home screen. Sadly, don't think we can do much about the flicker aside from that as it is redirecting in JavaScript, so it first loads the index page before the redirect. I've tried putting a 'holdReady' in, to try and delay it displaying the HTML, but it hasn't worked

**Comment by behaghel** [comment]
Cool. Last weird bit is the styling of the navigation: only on Blog the flickering makes the indicator of active tab (dark blue rule) to disappear. I will still merge but that would be nice to fix it ASAP

**Pull request merged**